src/Symfony/Component/Console/Attribute/Option.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Attribute;
+
+use Symfony\Component\Console\Completion\CompletionInput;
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Option
+{
+    private ?int $mode = null;
+    private string $typeName = '';
+
+    /**
+     * Represents a console command --option definition.
+     *
+     * If unset, the `name` and `default` values will be inferred from the parameter definition.
+     *
+     * @param array|string|null                                              $shortcut        The shortcuts, can be null, a string of shortcuts delimited by | or an array of shortcuts
+     * @param scalar|array|null                                              $default         The default value (must be null for self::VALUE_NONE)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public array|string|null $shortcut = null,
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
+    ) {
+        if (\is_string($suggestedValues) && !\is_callable($suggestedValues)) {
+            throw new \TypeError(\sprintf('Argument 5 passed to "%s()" must be either an array or a callable-string.', __METHOD__));
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public static function tryFrom(\ReflectionParameter $parameter): ?self
+    {
+        /** @var self $self */
+        if (null === $self = ($parameter->getAttributes(self::class, \ReflectionAttribute::IS_INSTANCEOF)[0] ?? null)?->newInstance()) {
+            return null;
+        }
+
+        $type = $parameter->getType();
+        $name = $parameter->getName();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The parameter "$%s" must have a named type. Untyped, Union or Intersection types are not supported for command options.', $name));
+        }
+
+        $self->typeName = $type->getName();
+
+        if (!\in_array($self->typeName, $allowedTypes = ['string', 'bool', 'int', 'float', 'array'], true)) {
```

Comment:
rather than assigning this variable within an if, for better visibility this could be a private static property of the class?

Reply:
added a constant instead 👍